### DISCUSSION TITLE
How to modify UI and functionality depending on user authorization rules?

### DISCUSSION THREAD

**Discussion by xperiandri** [unoplatform/uno#16948] [Q&A]
In ASP.NET Core and Blazor we have `Authorize` attribute and `IAuthorizationService` which allow to display or hide of some pieces of UI depending on the user role, or policy, or permission.  `IAuthorizationService` also allows performing authorization on a particular resource (object)

What approach can I use in Uno Platform?

For instance, I have a `NavigationView` and its items bust be different for different role

**Comment by agneszitte** [comment]
cc @jeromelaban, @nickrandolph

**Reply by Mike-E-angelo** [reply]
Hi @jeromelaban what is the point of grabbing all these claims/roles from the authentication service if they are not able to be easily applied in any way to the user interface?  I understand they can be sent to the external resource(s) (i.e. API) for the `Authorize` attribute/policy to handle there, but before that happens user interface elements should be displayed based on whether the user has the necessary claims to do so.  It's surprising to see that this area isn't addressed -- this thread is the best I could find and there's no obvious documentation I found around this area. 😞

**Comment by kazo0** [comment]
@xperiandri just a quick thought, maybe using a Converter that converts the Role to `Visibility.Visible` or `Visiblity.Collapsed` and use that on the Visibility properties of the elements that need to be invisible/visible

**Comment by kazo0** [comment]
Or maybe an Attached Property that you set on the elements in question. Something like `RequiredRoles="Admin, Editor, etc."` and the logic for when the property value changes can call the `IAuthorizationService`

Then set the Visibility of the element that has RequiredRoles attached depending on if the Role is included in the defined list

**Comment by Mike-E-angelo** [comment]
I would really like to know how this is done as well.  It appears that there are zero established components around this and it's rather unclear what the recommended pattern is to hide/show interface elements based on the user's claims/scopes/roles/etc. 🤔
